src/GovUk.Education.ExploreEducationStatistics.Content.Api/Controllers/ThemeController.cs

```diff
@@ -14,50 +14,58 @@ namespace GovUk.Education.ExploreEducationStatistics.Content.Api.Controllers
     [Produces(MediaTypeNames.Application.Json)]
     public class ThemeController : ControllerBase
     {
-        private readonly IBlobStorageService _blobStorageService;
+        private readonly IFileStorageService _fileStorageService;
 
-        public ThemeController(IBlobStorageService blobStorageService)
+        public ThemeController(IFileStorageService fileStorageService)
         {
-            _blobStorageService = blobStorageService;
+            _fileStorageService = fileStorageService;
         }
 
         [HttpGet("themes")]
         public async Task<ActionResult<IEnumerable<ThemeTree<PublicationTreeNode>>>> GetThemes()
         {
-            return await this.JsonContentResultAsync<IEnumerable<ThemeTree<PublicationTreeNode>>>(
-                () =>
-                    _blobStorageService.DownloadBlobText(
-                        PublicContentContainerName,
-                        PublicContentPublicationsTreePath()
-                    ),
-                NoContent()
-            );
+            try
+            {
+                return await _fileStorageService
+                    .GetDeserialized<IEnumerable<ThemeTree<PublicationTreeNode>>>(PublicContentPublicationsTreePath())
+                    .HandleFailuresOrOk();
+            }
+            catch (JsonException)
+            {
+                return new NoContentResult();
+            }
         }
 
         [HttpGet("download-themes")]
         public async Task<ActionResult<IEnumerable<ThemeTree<PublicationDownloadTreeNode>>>> GetDownloadThemes()
         {
-            return await this.JsonContentResultAsync<IEnumerable<ThemeTree<PublicationDownloadTreeNode>>>(
-                () =>
-                    _blobStorageService.DownloadBlobText(
-                        PublicContentContainerName,
+            try
+            {
+                return await _fileStorageService
+                    .GetDeserialized<IEnumerable<ThemeTree<PublicationDownloadTreeNode>>>(
                         PublicContentDownloadTreePath()
-                    ),
-                NoContent()
-            );
+                    )
+                    .HandleFailuresOrOk();
+            }
+            catch (JsonException)
```

Comment:
Feels to me like this should still be a runtime exception causing a 500 error.

Reply:
Yeh so I've only converted this to a `NoContentResult` because the original implementation (for the 'tree' endpoints) that I refactored this from did exactly that.

I agree that it should just throw a 500 though, so will remove the conversions to `NoContentResults`.